sw/device/lib/runtime/pmp.c

```diff
@@ -37,145 +37,100 @@ typedef enum pmp_csr_access_type {
   kPmpCsrAccessTypeWrite,
 } pmp_csr_access_type_t;
 
-// Because CSRs are encoded into the instructions, `pmpcfg` and `pmpaddr` cannot
-// be runtime values (must be constexpr). This set of macros allows to read
-// and write `pmpcfg` and `pmpaddr` CSRs in a common way. The alternative would
-// be to create an access function for every `pmpcfg` and `pmpaddr`
-// (20 or 40 together, dependent on the implementation).
-#define PMP_CSR_WRITE(csr, var) asm volatile("csrw " #csr ", %0;" : : "r"(var))
-
-#define PMP_CSR_READ(csr, var) asm volatile("csrr %0, " #csr ";" : "=r"(var) :)
-
-#define PMP_CSR_RW(access, csr, var)       \
-  do {                                     \
-    uint32_t csr_value;                    \
-    if (access == kPmpCsrAccessTypeRead) { \
-      PMP_CSR_READ(csr, csr_value);        \
-      *var = csr_value;                    \
-    } else {                               \
-      csr_value = *var;                    \
-      PMP_CSR_WRITE(csr, csr_value);       \
-    }                                      \
-  } while (false)
-
 static const bitfield_field32_t kPmpCfgModeField = {
     .mask = PMP_CFG_CSR_MODE_MASK, .index = PMP_CFG_CSR_A,
 };
 
 /**
- * Reads/writes to a `pmpcfgN` CSR.
- *
- * `N` is derived from a `region` (a single `pmpcfg` CSR contains configuration
- * information for `PMP_CFG_FIELDS_PER_REG` regions).
+ * Reads the pmpcfg for a given region.
  *
- * @param region PMP region ID to get/set.
- * @param access CSR access type read/write.
- * @param value Read value from a CSR, or value to write into a CSR.
- * @return `pmp_region_configure_result_t`.
+ * @param region PMP Region ID to read.
+ * @param[out] value Where to put the result of the read.
+ * @return `true` if `pmp_region_index_t` is valid and value was read, `false`
+ * otherwise.
  */
 PMP_WARN_UNUSED_RESULT
-static pmp_region_configure_result_t pmp_cfg_csr_rw(
-    pmp_region_index_t region, pmp_csr_access_type_t access, uint32_t *value) {
+static bool pmp_cfg_csr_read(pmp_region_index_t region, uint32_t *value) {
   switch (region) {
-    case 0:
-    case 1:
-    case 2:
-    case 3:
-      PMP_CSR_RW(access, pmpcfg0, value);
-      break;
-    case 4:
-    case 5:
-    case 6:
-    case 7:
-      PMP_CSR_RW(access, pmpcfg1, value);
-      break;
-    case 8:
-    case 9:
-    case 10:
-    case 11:
-      PMP_CSR_RW(access, pmpcfg2, value);
-      break;
-    case 12:
-    case 13:
-    case 14:
-    case 15:
-      PMP_CSR_RW(access, pmpcfg3, value);
-      break;
+#define PMP_REGION(region_id, config_reg_id)                               \
```

Comment:
Do you think we should indent these preprocessor directives? After all, the `#include` below actually expands to something that actually makes both behave as if they were scoped.

Reply:
I think the indent style for these directives is weird:

```
#    define PMP_REGION
```

I'm kind of indifferent, to be honest, given that the body indent and define-scope indent are different.

Reply:
It's common to only indent preprocessor directives with respect to other preprocessor directives (specifically `#if`s), rather than also with surrounding code. I realise these are "scoped", but that scoping is fairly implicit to the preprocessor, and this is what clang-format asked me to do.